android/KMAPro/build.sh

```diff
@@ -44,97 +80,62 @@ function makeLocalSentryRelease() {
   sentry-cli releases finalize "$VERSION_GIT_TAG"
 }
 
-NO_DAEMON=false
-ONLY_DEBUG=false
-DO_KEYBOARDS_DOWNLOAD=false
-DO_MODELS_DOWNLOAD=false
-DO_SENTRY_LOCAL_UPLOAD=false
 
-# Parse args
-while [[ $# -gt 0 ]] ; do
-    key="$1"
-    case $key in
-        -no-daemon)
-            NO_DAEMON=true
-            ;;
-        -upload-sentry)
-            # Overrides default set by build-utils.sh
-            UPLOAD_SENTRY=true
-            ;;
-        -debug)
-            ONLY_DEBUG=true
-            ;;
-        -download-resources)
-            DO_KEYBOARDS_DOWNLOAD=true
-            DO_MODELS_DOWNLOAD=true
-            ;;
-        -h|-\?)
-            display_usage
-            ;;
-    esac
-    shift # past argument
-done
-
-KEYBOARD_PACKAGE_ID="sil_euro_latin"
-KEYBOARDS_TARGET="$KEYMAN_ROOT/android/KMAPro/kMAPro/src/main/assets/${KEYBOARD_PACKAGE_ID}.kmp"
-MODEL_PACKAGE_ID="nrc.en.mtnt"
-MODELS_TARGET="$KEYMAN_ROOT/android/KMAPro/kMAPro/src/main/assets/${MODEL_PACKAGE_ID}.model.kmp"
-
-# Verify default keyboard and dictionary exist
-if [[ ! -f "$KEYBOARDS_TARGET" ]]; then
-  echo "$KEYBOARDS_TARGET doesn't exist. Will download the latest version"
-  DO_KEYBOARDS_DOWNLOAD=true
-fi
+#### Build action definitions ####
 
-if [[ ! -f "$MODELS_TARGET" ]]; then
-  echo "$MODELS_TARGET doesn't exist. Will download the latest version"
-  DO_MODELS_DOWNLOAD=true
-fi
+# Check about cleaning artifact paths and upload directories
+if builder_start_action clean; then
+  cd "$KEYMAN_ROOT/android/KMAPro/"
```

Comment:
Is this necessary? You've already `cd "$THIS_SCRIPT_PATH"` earlier (and perhaps if you change folders elsewhere, such as in `convert_markdown_to_html()`, you should restore path on exit with `pushd`/`popd`).

Furthermore, every step within this action has full path references.

```suggestion
```